lib/ancestry/class_methods.rb

```diff
@@ -29,30 +29,25 @@ def orphan_strategy= orphan_strategy
 
     # Arrangement
     def arrange options = {}
-      # Get all nodes ordered by ancestry and start sorting them into an empty hash
-      arrange_nodes self.ancestry_base_class.reorder(options.delete(:order)).where(options)
+      # Get all nodes and start sorting them into an empty hash
+      scope = self.ancestry_base_class
+      if (order = options.delete(:order))
+        scope = scope.order(order)
+      end
+      arrange_nodes scope.where(options)
     end
 
     # Arrange array of nodes into a nested hash of the form
     # {node => children}, where children = {} if the node has no children
+    # If a node's parent is not included, the node will be included as if it is a top level node
     def arrange_nodes(nodes)
-      arranged = ActiveSupport::OrderedHash.new
-      min_depth = Float::INFINITY
-      index = Hash.new { |h, k| h[k] = ActiveSupport::OrderedHash.new }
```

Comment:
I remember the OrderedHash being important for some reason...can't remember why.

Reply:
Or maybe I was just being overly focused on consistency with the old code.

Reply:
A Hash is ordered since ruby 1.9 (?)

I think it is deprecated

https://api.rubyonrails.org/classes/ActiveSupport/OrderedHash.html

Reply:
Ah I see we've already removed support for Ruby 1.9 anyway, so this is fine :+1:

Reply:
good call - I forgot that this change was recent.